From 2d28901bb2eac1329e935b0d7f8418a27c0b0067 Mon Sep 17 00:00:00 2001
From: Chris Feist <cfeist@redhat.com>
Date: Tue, 22 Sep 2015 17:19:37 -0500
Subject: [PATCH] Fix for crm_node -l output change

- crm_node -l now outputs a status after the node id and node name we
  now ignore lines where the 3rd field is "lost".
---
 pcs/utils.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/pcs/utils.py b/pcs/utils.py
index 0b8d03f..88362b3 100644
--- a/pcs/utils.py
+++ b/pcs/utils.py
@@ -1706,8 +1706,9 @@ def getPacemakerNodesID(allow_failure=False):
 
     pm_nodes = {}
     for line in output.rstrip().split("\n"):
-        node_info = line.rstrip().split(" ",1)
-        pm_nodes[node_info[0]] = node_info[1]
+        node_info = line.rstrip().split(" ")
+        if len(node_info) <= 2 or node_info[2] != "lost":
+            pm_nodes[node_info[0]] = node_info[1]
 
     return pm_nodes
 
-- 
1.9.1

